authors/apps/authentication/serializers.py

```diff
@@ -16,20 +16,63 @@ class RegistrationSerializer(serializers.ModelSerializer):
 
     # Ensure passwords are at least 8 characters long, no longer than 30
     # Password has no spaces
-    password = serializers.RegexField(
-        regex=("^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)\S{8,}$"),
-        max_length=30,
-        min_length=8,
+    password = serializers.CharField(
+        max_length=250,
         write_only=True,
-        required=True,
-        error_messages={
-            'required': 'Password is a required field',
-            'min_length': 'Password must be at least 8 characters long',
-            'max_length': 'Password cannot be more than 30 characters',
-            'invalid': 'Password must have alphanumeric characters with no space',
-        }
+        allow_null=True,
+        allow_blank=True,
+        required=True
     )
 
+    def validate(self, value):
+        self.get_password_field_erros(value)
+        try:
+            validate_password(value.get("password"))
+        except ValidationError as e:
+            raise serializers.ValidationError({
+                "password": list(e)
+            })
+        self.get_password_policy_rrors(value.get("password"))
+        return value
+
+    def get_password_policy_rrors(self, password):
+        """
+        Method to captures password policy errors
+        """
+        stats = PasswordStats(password)
+        errors = []
+        if stats.letters_uppercase < 2:
+            errors.append("password should have at least of 2 uppercase letters")
+
+        if stats.letters_lowercase < 3:
+            errors.append("password should have at least 3 lowercase letters")
+
+        if stats.numbers < 2:
+            errors.append("password should have at least 2 digits")
+
+        if stats.special_characters < 1:
+            errors.append("password should have at least 1 special character")
+
+        if stats.sequences_length > 2:
+            errors.append("password should not have a repeating sequence")
+
+        if stats.repeated_patterns_length > 2:
+            errors.append("password should not have a repeating characters")
+            
+        if errors:
+            raise serializers.ValidationError({
+                "password": list(errors)
+            })
+
+    def get_password_field_erros(self, value):
+        """
+        Method to get null, blank and empty fields for password
+        """
+        if not value.get("password"):
+            raise serializers.ValidationError({
+                "password": "Provide your password"
```

Comment:
This part is also untested. Would you test all the new code that this PR is adding to the codebase

Reply:
I have tested it now